lib/customer.rb

```diff
@@ -0,0 +1,46 @@
+require_relative '../lib/order'
+require 'csv'
+
+class Customer
+  @@customers = []
+
+  attr_reader :id
+  attr_accessor :email, :address
+  ID = 0
+  EMAIL = ""
+  ADDRESS = {}
+
+  def initialize(id, email, address)
+    @id = id
+    @email = email
+    @address = address
+  end
+
+  def self.all
+    @@customers = CSV.open('data/customers.csv', 'r').map do |row|
+      id = row[0].to_i
+      email = row[1]
+      address = {
+        :street => row[2],
+        :city => row[3],
+        :state => row[4],
+        :zip => row[5]
+      }
+      Customer.new(id, email, address)
+    end
+    return @@customers
+  end
+
+  def self.find(id)
+    x = ""
+    Customer.all.each do |cust|
+      if cust.id == id
+        x = cust
+        break
+      else
+        x = nil
+      end
+    end
+    return x
+  end
```

Comment:
I think this method would be better if you used a variable name more descriptive than `x`!

Also, this method ends up doing the right thing, but could be cleaned up a little bit. Does `x` need to be assigned an empty string `""` in the beginning?

Using a `.each` on `Customer.all` works, but maybe there's a fancy `Enumerable` method we can use, like `.find` or something :)